lang/java/avro/src/main/java/org/apache/avro/Schema.java

```diff
@@ -1295,8 +1295,7 @@ private static class FixedSchema extends NamedSchema {
 
     public FixedSchema(Name name, String doc, int size) {
       super(Type.FIXED, name, doc);
-      if (size < 0)
-        throw new IllegalArgumentException("Invalid fixed size: " + size);
+      SystemLimitException.checkMaxBytesLength(size);
```

Comment:
You can also set the field using the method return value

Reply:
Oh pardon, I missed this comment between rebasing and cherry-picking!  I don't think it's a big deal, but I'll pick this up in the next change to Schema.  Thanks!